### PR TITLE
PROD-3145 avatar dropdown -> master

### PR DESCRIPTION
**Pull request by vas3a**
https://topcoder.atlassian.net/browse/PROD-3145

Adds avatar dropdown & refactors all the code related to the PopupMenu in it's own component so it can be reused

### CODE REVIEW

File: src/lib/components/PopupMenu.svelte [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    // store when the popup is about to hide because user hovered outside
+    let willHide: boolean;
+
+    // listen to mouse over events on the entire document
+    // when the user is outside the target element, we hide the popup
+    document.addEventListener('mouseover', async function mc(ev) {
+      // check to see if user is still hovering the targetEl or the popup
+      const isStillHovering = wraps.some(w => w.contains(ev.target as Node))
+      // if user isn't hoverint the target/popup, the popup is about to hide
```
Comment (testflyjets) [inline src/lib/components/PopupMenu.svelte:40]: minor typo: `hoverint` => `hovering`

File: src/lib/components/PopupMenu.svelte [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      willHide = !isStillHovering;
+
+      // do nothing if user is still hovering
+      if (isStillHovering) {
+        return
+      }
+
+      // provide a delay before taking any action (better UX)
+      await delay(200)
```
Comment (testflyjets) [inline src/lib/components/PopupMenu.svelte:49]: Just curious if 200ms is considered a standard for this type of UX "debounce"?

**Review by testflyjets — APPROVED** [review]

File: src/lib/components/PopupMenu.svelte [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    if (!el) {
+      return
+    }
+
+    el.dataset.targetKey = undefined
+    el.removeEventListener('mouseenter', handleMouseover)
+  }
+  
+  // each time the targetEl changes, 
+  // call bindEvents to lisent for mouse events
```
Comment (brooketopcoder) [inline src/lib/components/PopupMenu.svelte:100]: typo `lisent` -> `listen`

File: src/lib/components/UserAvatar.svelte [hunk 4]
```diff
@@ -1,16 +1,40 @@
...
+</div>
+
+<PopupMenu targetEl={elRef} bind:isVisible={popupIsVisible}>
+  <div class={styles.avatarDropdownMenu}>
+    <ul>
+      <li>
+        <a href={`/members/${user.handle}`}>My Profile</a>
+      </li>
+      <li>
+        <a href="/settins/profile">Settings</a>
```
Comment (brooketopcoder) [inline src/lib/components/UserAvatar.svelte:33]: typo `settins` -> `settings`

File: src/lib/marketing-navigation/components/NavigationBar.svelte [hunk 5]
```diff
@@ -21,15 +24,20 @@
     menuItems={menuItems}
     activeRoute={activeRoute}
     bind:hoveredMenuItem={hoveredMenuItem}
+    bind:hovereElement={hoveredElement}
```
Comment (brooketopcoder) [inline src/lib/marketing-navigation/components/NavigationBar.svelte:27]: typo `hovere` -> `hovered`

**Pull request merged**